operators/src/processing/reprojection.rs

```diff
@@ -191,24 +204,263 @@ where
+            spatial_reference: self.params.target_spatial_reference.into(),
+            data_type: in_desc.data_type,
+            measurement: in_desc.measurement.clone(),
+        };
+
+        let state = RasterReprojectionState {
+            source_srs: Option::from(in_desc.spatial_reference).unwrap(),
+            target_srs: self.params.target_spatial_reference,
+            tiling_spec: context.tiling_specification(),
+        };
+
+        Ok(
+            InitializedRasterReprojection::new(out_desc, initialized_raster_sources, vec![], state)
+                .boxed(),
+        )
+    }
+}
+
+impl InitializedOperator<RasterResultDescriptor, TypedRasterQueryProcessor>
+    for InitializedRasterReprojection
+{
+    // i know there is a macro somewhere. we need to re-work this when we have the no-data value anyway.
+    #[allow(clippy::clippy::too_many_lines)]
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let q = self.raster_sources[0].query_processor()?;
+
+        let s = self.state;
+
+        Ok(match self.result_descriptor.data_type {
+            geoengine_datatypes::raster::RasterDataType::U8 => {
+                let qt = q.get_u8().unwrap();
+                TypedRasterQueryProcessor::U8(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
+                    s.tiling_spec,
+                    0_u8,
+                )))
+            }
+            geoengine_datatypes::raster::RasterDataType::U16 => {
+                let qt = q.get_u16().unwrap();
+                TypedRasterQueryProcessor::U16(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
+                    s.tiling_spec,
+                    0_u16,
+                )))
+            }
+
+            geoengine_datatypes::raster::RasterDataType::U32 => {
+                let qt = q.get_u32().unwrap();
+                TypedRasterQueryProcessor::U32(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
+                    s.tiling_spec,
+                    0_u32,
+                )))
+            }
+            geoengine_datatypes::raster::RasterDataType::U64 => {
+                let qt = q.get_u64().unwrap();
+                TypedRasterQueryProcessor::U64(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
+                    s.tiling_spec,
+                    0_u64,
+                )))
+            }
+            geoengine_datatypes::raster::RasterDataType::I8 => {
+                let qt = q.get_i8().unwrap();
+                TypedRasterQueryProcessor::I8(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
```

Comment:
Isn't there a macro for that?

Reply:
yes there is a macro but it is not easy to get the code to compile and i would prefer to change this to a macro when all issues are solved

Reply:
would still avoid this until the NO DATA stuff is handled